src/main/java/org/takes/facets/fork/MediaType.java

```diff
@@ -108,12 +75,73 @@ public int compareTo(final MediaType type) {
      */
     public boolean matches(final MediaType type) {
         final String star = "*";
-        return (this.high.equals(star)
-            || type.high.equals(star)
-            || this.high.equals(type.high))
-            && (this.low.equals(star)
-            || type.low.equals(star)
-            || this.low.equals(type.low));
+        return (this.getHigh().equals(star)
+            || type.getHigh().equals(star)
+            || this.getHigh().equals(type.getHigh()))
+            && (this.getLow().equals(star)
+            || type.getLow().equals(star)
+            || this.getLow().equals(type.getLow()));
+    }
+
+    /**
+     * Get the parts.
+     * @return The parts of the media type.
+     */
+    @Cacheable(forever = true)
+    private String[] getParts() {
+        return this.text.split(";", 2);
+    }
+
+    /**
+     * Get the priority.
+     * @return The priority of the media type.
+     */
+    @Cacheable(forever = true)
+    private Double getPriority() {
```

Comment:
@dalifreire please read http://www.yegor256.com/2014/09/16/getters-and-setters-are-evil.html